### PR TITLE
[maskedtensor] add docs

### PR DESCRIPTION
**Pull request by george-qi**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #84887

cc @nikitaved @pearu @cpuhrsch @amjames @bhosmer

### CODE REVIEW

File: docs/source/masked.rst [hunk 1]
```diff
@@ -1,11 +1,291 @@
...
+    positive
+    signbit
+    isnan
+
+Binary Operators
+----------------
+
+As you may have seen in the tutorial, :class:`MaskedTensor` also has binary operations implemented with the caveat
+that the masks in the two MaskedTensors must match or else an error will be raised. As noted in the error, if you
+need support for a particular operator or have proposed semantics for how they should be behave instead, please open
```
Comment (cpuhrsch) [inline docs/source/masked.rst:157]: nit: "for how they should be behave instead" -> typos?

File: docs/source/masked.rst [hunk 2]
```diff
@@ -1,11 +1,291 @@
...
+    :nosignatures:
+
+    logaddexp
+    logaddexp2
+    equal
+    fmin
+    minimum
+    fmax
+
+Reductions
```
Comment (cpuhrsch) [inline docs/source/masked.rst:217]: I think we can add a section to this as well as a follow up.

**Comment by cpuhrsch** [comment]
Lint issues seem real. You can address them using https://github.com/pytorch/pytorch/wiki/lintrunner

File: docs/source/masked.rst (hunk 2 above)
Comment (george-qi) [reply docs/source/masked.rst:217]: elaborated

**Comment by george-qi** [comment]
Somehow, embedding links into ordered/unordered lists in RST is quite finicky/difficult.. e.g.

```
- `Overview <link>`_ -- insert description here
```

doesn't really work, because it yields a mix of warnings like:

`Inline interpreted text or phrase reference start-string without end-string.` and `Mismatch: both interpreted text role prefix and reference suffix.`

I could only get the warnings to go away if the entire bullet point consisted of the link, i.e.

```
- `Overview -- insert description here`_

.. _Overview -- insert description here: <link>
```

which seems to be used in examples like: https://github.com/pytorch/pytorch/blob/master/docs/source/fx.rst#graph-manipulation-examples. If anyone has anything more graceful, let me know -- tried this for a while

**Review by cpuhrsch — APPROVED** [review]

**Pull request closed without merging**